metaflow/decorators.py

```diff
@@ -374,12 +373,17 @@ def _base_flow_decorator(decofunc, *args, **kwargs):
         if isinstance(cls, type) and issubclass(cls, FlowSpec):
             # flow decorators add attributes in the class dictionary,
             # _flow_decorators.
-            if decofunc.name in cls._flow_decorators:
+            if decofunc.name in cls._flow_decorators and not decofunc.allow_multiple:
```

Comment:
I am a little confused as to why step and flow decorators are treated differently wrt to the allow_multiple.